### PR TITLE
Refactor Utterance Services to Return Response Objects

### PR DESCRIPTION
**Pull request by ashleygranitto**
I added three Response objects: StatusResponse, ErrorResponse, & UrlResponse (the latter two which extend StatusResponse). Instead of having the Utterance servlets return raw JSON, I instead return a StatusResponse object that has been converted to JSON via GSON. I updated the servlet testing files to reflect this update.

### CODE REVIEW

File: server/src/main/java/com/google/speech/tools/voxetta/servlets/BlobstoreLinkServlet.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  private UtteranceService service = new DatastoreUtteranceService(); 
+ 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    // Create and return a Blobstore Upload URL  
+    try {
+      String uploadUrl = service.getFormUrl();
+      String successJson = StatusResponse.convertToJson(new UrlResponse(true, uploadUrl));
```
Comment (mmallett) [inline server/src/main/java/com/google/speech/tools/voxetta/servlets/BlobstoreLinkServlet.java:45]: If you remove the static from the definition, you could use the `this` instance and be a bit more expressive. Would read like

```java
String json = new UrlResponse(true, uploadUrl).toJson();
```

File: server/src/main/java/com/google/speech/tools/voxetta/data/StatusResponse.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+   * @param success Represents if a backend fetch was successful or not. 
+   */ 
+  public StatusResponse(boolean success) {
+    this.success = success;
+  }
+
+  /** 
+   * Returns a JSON representation of a StatusResponse.  
+   */
+  public static String convertToJson(StatusResponse response) {
```
Comment (mmallett) [inline server/src/main/java/com/google/speech/tools/voxetta/data/StatusResponse.java:38]: If you remove the static, you can return a json of `this`

```java
public string toJson() {
  return new Gson.toJson(this);
}
```

And call it like

```java
string json = new StatusResponse(true).toJson();
```

File: server/src/main/java/com/google/speech/tools/voxetta/servlets/BlobstoreLinkServlet.java (hunk 1 above)
Comment (ashleygranitto) [reply server/src/main/java/com/google/speech/tools/voxetta/servlets/BlobstoreLinkServlet.java:45]: This is much better--thank you!! adjusted

File: server/src/main/java/com/google/speech/tools/voxetta/data/StatusResponse.java (hunk 2 above)
Comment (ashleygranitto) [reply server/src/main/java/com/google/speech/tools/voxetta/data/StatusResponse.java:38]: adjusted

**Pull request merged**